File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    width = 0
    high = 0
    for w, h in sizes:
        w1 = max(width, w)
        h1 = max(high, h)
        w2 = max(width, h)
        h2 = max(high, w)
        
        if w1 * h1 < w2 * h2:
            width = w1
            high = h1
        else:
            width = w2
            high = h2
            
    return width * high

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int maxW = 0;
        int maxH = 0;
        for (int[] size: sizes) {
            // 긴 쪽이 가로, 짧은 쪽이 세로 고정
            // 가로, 세로 각각 최대 길이 저장
			maxW = Math.max(Math.max(size[0], size[1]), maxW);
            maxH = Math.max(Math.min(size[0], size[1]), maxH);              
        }
        return maxW * maxH;
    }
}